play-json/shared/src/main/scala/Reads.scala

```diff
@@ -398,8 +398,57 @@ trait DefaultReads extends LowPriorityDefaultReads {
   /**
    * Deserializer for Array[T] types.
    */
-  implicit def ArrayReads[T: Reads: ClassTag]: Reads[Array[T]] = new Reads[Array[T]] {
-    def reads(json: JsValue) = json.validate[List[T]].map(_.toArray)
+  implicit def ArrayReads[T: Reads: ClassTag]: Reads[Array[T]] =
+    Reads[Array[T]] { _.validate[List[T]].map(_.toArray) }
+
+  // If the number is decimal, considering it's /1000 of millis,
+  // otherwise considering it's directly in millis
+  private def numberDuration(n: BigDecimal): FiniteDuration = {
```

Comment:
It would be nice to separate the number stuff to something like `millisDurationReads` and `millisDurationWrites`. If you want to support both in the `Reads` you could use `orElse`